lib/manager/gomod/artifacts.ts

```diff
@@ -116,25 +114,43 @@ export async function updateArtifacts({
       logger.debug('Removed some relative replace statements from go.mod');
     }
     await writeLocalFile(goModFileName, massagedGoMod);
+    // array of GOMODULE registries which should be fetched from
+    // passed to go as GOPRIVATE
+    let registryUrls: string[] = [];
+
+    // add GOPRIVATE environment variables
+    if (process.env.GOPRIVATE) {
+      const goPrivates = process.env.GOPRIVATE.split(',');
+      registryUrls = registryUrls.concat(goPrivates);
+    }
+
+    // add explicit registryUrls
+    if (config.registryUrls) {
+      registryUrls = registryUrls.concat(config.registryUrls);
```

Comment:
These urls are probably full urls. So do a foreach and check if it's full.

* true: push to `registryUrls`, parse to get domain and push to `goPrivate`.
* false: push to `goPrivate`, make full and push to `registryUrls`